### PR TITLE
I119 cleanup and logging

### LINKED ISSUE #119: EWUTeam log all exceptions, cleanup
**Describe the issue**

There are non-logged exceptions.  
There are compiler warnings in the project.

**To Reproduce**:

N/A, must review code.

**Expected behavior**:

All exceptions logged.

**Actual behavior**:

Not all exceptions logged.

**Environment**:

All.

**Log Info**:

None.

**Screenshots**:

None.

**Additional context**:

None.

### PR DESCRIPTION
**Pull request by lane55**
### Description of what the PR does

Adds logging and cleansup warnings

### Issue which the PR fixes

#119

### Environment in which the PR was developed (OS,IDE, Java version, etc.)

All

### Precise steps for _testing_ the PR (i.e., how to demonstrate that it works correctly)

build project, no warning should be listed.

For logging, check all catch blocks for logged exceptions.

### CODE REVIEW

File: projects/EWTeam/src/ServerInterface.java [hunk 1]
```diff
@@ -133,6 +130,7 @@ private ServerInterface() {
 			StaticLog.info("Found value for scoreboard2password in pc2v9.ini");
 			pc2Properties = null;
 		} catch (Exception e) {
+			StaticLog.unclassified("Error loading pc2v9.ini "+ e.getMessage(), e);
```
Comment (clevengr) [inline projects/EWTeam/src/ServerInterface.java:133]: It does not seem to me that "unclassified" is an appropriate log message type for this condition. Failure to load a pc2v9.ini with a scoreboard password seems like it should be a "warning".

File: src/edu/csus/ecs/pc2/core/InternalController.java [hunk 2]
```diff
@@ -1002,6 +1002,8 @@ public void initializeServer(IInternalContest inContest) throws IOException, Cla
             
             handleCDPLoad(true);
             
+            loadDefaultJudgementsIfMissing();
```
Comment (clevengr) [inline src/edu/csus/ecs/pc2/core/InternalController.java:1005]: The total description of the Issue (#119) to which this PR applies says:
`There are non-logged exceptions.
There are compiler warnings in the project.
`
It does not seem to me that addition of a whole new method, which has potentially significant logic ramifications, falls under Issue #119.  This should have been submitted as a separate issue/PR.

File: src/edu/csus/ecs/pc2/core/InternalController.java [hunk 3]
```diff
@@ -1058,23 +1060,58 @@ private void insureProfileDirectory(Profile profile) {
...
     }
+    
+    /**
+     * Load judgements from reject.ini
+     * @param contest
+     * @param cdpConfigDirectory
+     * @return
+     * @throws Exception
+     */
+    protected boolean loadJudgements(IInternalContest contest, File cdpConfigDirectory) throws Exception {
```
Comment (clevengr) [inline src/edu/csus/ecs/pc2/core/InternalController.java:1071]: Maybe I'm missing something about how the PR review process works, but as stated in the previous review comment this looks like it is adding a whole new method which has nothing to do with Issue #119.

File: src/edu/csus/ecs/pc2/core/InternalController.java [hunk 4]
```diff
@@ -1058,23 +1060,58 @@ private void insureProfileDirectory(Profile profile) {
...
 
         if (!isContactingRemoteServer()) {
 
             if (contest.getJudgements().length == 0) {
-
                 if (loadedJudgementsFromIni(Constants.JUDGEMENT_INIT_FILENAME)) {
                     info("Loaded judgements from " + Constants.JUDGEMENT_INIT_FILENAME);
-                } else {
-                    info(Constants.JUDGEMENT_INIT_FILENAME + " not found.  Loading default judgements");
-                    loadDefaultJudgements();
```
Comment (clevengr) [inline src/edu/csus/ecs/pc2/core/InternalController.java:1072]: Again, unless I'm misunderstanding the PR review process/procedure/mechanisms, this looks like it is a logic change (removal of the call to `loadDefaultJudgements()` under the `else` condition), which does not seem to me to have anything to do with the purpose of Issue #119.  This should be submitted under a separate issue/PR.

File: src/edu/csus/ecs/pc2/core/InternalController.java [hunk 5]
```diff
@@ -1058,23 +1060,58 @@ private void insureProfileDirectory(Profile profile) {
...
-                }
+                } 
             }
         }
     }
 
+    /**
+     * Load default judgements.
+     */
+    protected void loadDefaultJudgementsIfMissing() {
```
Comment (clevengr) [inline src/edu/csus/ecs/pc2/core/InternalController.java:1103]: As noted in previous review comments, this does not look like it has anything to do with Issue #119.

File: src/edu/csus/ecs/pc2/core/InternalController.java [hunk 6]
```diff
@@ -1058,23 +1060,58 @@ private void insureProfileDirectory(Profile profile) {
...
+        if (!isContactingRemoteServer()) {
+
+            if (contest.getJudgements().length == 0) {
+                info(Constants.JUDGEMENT_INIT_FILENAME + " not found.  Loading default judgements");
+                loadDefaultJudgements();
+            }
+        }
+    }
+
+
```
Comment (clevengr) [inline src/edu/csus/ecs/pc2/core/InternalController.java:1114]: Again, the entire logic block above is unrelated to the stated purpose of Issue #119 and PR #120.

File: src/edu/csus/ecs/pc2/core/InternalController.java [hunk 7]
```diff
@@ -2589,6 +2626,12 @@ protected void handleCDPLoad(boolean allowedToLoadConfig) {
                     
                     File cdpConfigDir = loader.findCDPConfigDirectory(new File(entryLocation));
                     
+                    if (contest.getJudgements().length == 0) {
+                        if (loadJudgements(contest, cdpConfigDir)){
+                            System.out.println("Loaded judgements from "+getCdpConfigRejectIniFilename(cdpConfigDir));
+                        }
+                    }
+
```
Comment (clevengr) [inline src/edu/csus/ecs/pc2/core/InternalController.java:2634]: Ditto the previous review comments; this does not appear to have anything to do with Issue #119 or PR #120.

File: src/edu/csus/ecs/pc2/core/report/JudgementReport.java [hunk 8]
```diff
@@ -14,16 +14,15 @@
...
  * 
  * @author pc2@ecs.csus.edu
- * @version $Id$
  */
 
-// $HeadURL$
-
 public class JudgementReport implements IReport {
 
+    public static final String NO_JUDGEMENT_PREFIX = "No - ";
```
Comment (clevengr) [inline src/edu/csus/ecs/pc2/core/report/JudgementReport.java:24]: As stated in previous review comments, this does not appear to have anything to do with the stated purpose of Issue #119 or PR #120.  Further, this could have _significant_ ramifications; there have already been issues with the concept of having the string "No - " as part of a judgement.  This should not be included in this PR.

File: src/edu/csus/ecs/pc2/core/report/JudgementReport.java [hunk 9]
```diff
@@ -67,6 +66,12 @@ public void writeReport(PrintWriter printWriter) {
             printWriter.print(hiddenText+" id=" + judgement.getElementId());
             printWriter.println();
         }
+        
+        printWriter.println();
+        printWriter.println("# reject.ini");
+        printWriter.println("#");
+        printRejectINI(printWriter, contest);
```
Comment (clevengr) [inline src/edu/csus/ecs/pc2/core/report/JudgementReport.java:73]: I think it's a bit of a stretch to say that addition of a whole new method which alters how a report appears falls under the category of the declared issue (#119), which is
`There are non-logged exceptions.
There are compiler warnings in the project.
`

File: src/edu/csus/ecs/pc2/core/report/JudgementReport.java [hunk 10]
```diff
@@ -76,6 +81,23 @@ public void printHeader(PrintWriter printWriter) {
         printWriter.println();
         printWriter.println(getReportTitle() + " Report");
     }
+    
+    public void printRejectINI(PrintWriter printWriter, IInternalContest inContest) {
```
Comment (clevengr) [inline src/edu/csus/ecs/pc2/core/report/JudgementReport.java:85]: See previous review comment.  Adding this method doesn't seem to me to fall under Issue #119.

File: projects/EWTeam/src/ServerInterface.java (hunk 1 above)
Comment (lane55) [reply projects/EWTeam/src/ServerInterface.java:133]: I agree and in effect unclassified is a warning - take a look at the unclassified method.
I assume unclassified was used to indicate that this log level has not been classified yet,
there are a number of other logs that are unclassified.   
You alao indicated that you thought it was a good idea to minimize the time spent on these changes.

File: src/edu/csus/ecs/pc2/core/InternalController.java (hunk 2 above)
Comment (lane55) [reply src/edu/csus/ecs/pc2/core/InternalController.java:1005]: I did not make this change.

File: src/edu/csus/ecs/pc2/core/InternalController.java (hunk 3 above)
Comment (lane55) [reply src/edu/csus/ecs/pc2/core/InternalController.java:1071]: I did not make this change.

File: src/edu/csus/ecs/pc2/core/InternalController.java (hunk 4 above)
Comment (lane55) [reply src/edu/csus/ecs/pc2/core/InternalController.java:1072]: I did not make this change.

File: src/edu/csus/ecs/pc2/core/InternalController.java (hunk 5 above)
Comment (lane55) [reply src/edu/csus/ecs/pc2/core/InternalController.java:1103]: I did not make this change.

File: src/edu/csus/ecs/pc2/core/InternalController.java (hunk 6 above)
Comment (lane55) [reply src/edu/csus/ecs/pc2/core/InternalController.java:1114]: I did not make this change.

File: src/edu/csus/ecs/pc2/core/report/JudgementReport.java (hunk 8 above)
Comment (lane55) [reply src/edu/csus/ecs/pc2/core/report/JudgementReport.java:24]: I did not make this change.

File: src/edu/csus/ecs/pc2/core/report/JudgementReport.java (hunk 9 above)
Comment (lane55) [reply src/edu/csus/ecs/pc2/core/report/JudgementReport.java:73]: I did not make this change.

File: src/edu/csus/ecs/pc2/core/report/JudgementReport.java (hunk 10 above)
Comment (lane55) [reply src/edu/csus/ecs/pc2/core/report/JudgementReport.java:85]: I did not make this change.

File: src/edu/csus/ecs/pc2/core/InternalController.java (hunk 2 above)
Comment (troy2914) [reply src/edu/csus/ecs/pc2/core/InternalController.java:1005]: seems your fork may have been out of date in that case.

**Comment by clevengr** [comment]
Based on the large number of commits in this PR that by Doug's own comments do not belong there, I'm closing/rejecting this PR.  Please resync your local clone and GitHub fork repos with the pc2ccs/pc2v9 repo (see https://github.com/pc2ccs/pc2v9/wiki/Development-Flow), rebase your issue branch on top of the (updated) Develop branch, and submit a new PR.
